brainiak/fcma/preprocessing.py

```diff
@@ -205,29 +83,26 @@ def _separate_epochs(activity_data, epoch_list):
     return raw_data, labels
 
 
-def prepare_fcma_data(data_dir, extension, epoch_file, mask_file1,
-                      mask_file2=None):
-    """ obtain the data for correlation-based computation and analysis
+def prepare_fcma_data(images, conditions, mask1, mask2=None,
+                      comm=MPI.COMM_WORLD):
+    """Prepare data for correlation-based computation and analysis.
 
-    read the data in and generate epochs of interests,
-    then broadcast to all workers
+    Generate epochs of interests, then broadcast to all workers.
 
     Parameters
     ----------
-    data_dir: str
-        the path to all subject files
-    extension: str
-        the file extension, usually nii.gz or nii
-    epoch_file: str
-        the absolute path of the epoch file
-    mask_file1: str
-        the absolute path of the mask file,
-        we apply the mask right after reading a file for saving memory.
-    mask_file2: Optional[str]
-        the absolute path of another mask file,
-        we apply the mask right after reading a file for saving memory.
-        If it is not specified, the method will assign None to
-        the returning variable raw_data2
+    images: Iterable[SpatialImage]
```

Comment:
`SpatialImage` is imported, so users can see what it means. Actually, Pyflakes would complain otherwise, which is a nice side-effect of using PEP 484 annotations. Furthermore, Sphinx shows the full name in the HTML documentation.